File: 백준/Bronze/17206. 준석이의 수학 숙제/준석이의 수학 숙제.java
import java.io.*;

class Main {
    static int nextInt() throws IOException {
        int c, n = System.in.read() & 15;
        while ((c = System.in.read()) > 32) {
            n = (n << 3) + (n << 1) + (c & 15);
        }
        return n;
    }

    public static void main(String[] args) throws IOException {
        int T = nextInt();
        StringBuilder sb = new StringBuilder();

        while (T-- > 0) {
            int N = nextInt();
            int k3 = N / 3, k7 = N / 7, k21 = N / 21;
            int sum = 0;
            sum += arithS(k3) * 3;
            sum += arithS(k7) * 7;
            sum -= arithS(k21) * 21;
            sb.append(sum).append('\n');
        }
        System.out.print(sb);
    }

    static int arithS(int k) {
        return k * (k + 1) / 2;
    }
}

File: 백준/Bronze/17206. 준석이의 수학 숙제/준석이의 수학 숙제.rs
use std::io::{self, Read};

fn main() {
    let mut buffer = String::new();
    io::stdin().read_to_string(&mut buffer).unwrap();
    let mut it = buffer.split_whitespace();

    let mut out = String::new();
    let t: usize = it.next().unwrap().parse().unwrap();
    for _ in 0..t {
        let n: i32 = it.next().unwrap().parse().unwrap();
        let k3: i32 = n / 3;
        let k7: i32 = n / 7;
        let k21: i32 = n / 21;
        let mut sum: i64 = 0;
        sum = sum + (arith_series(k3) * 3) as i64;
        sum = sum + (arith_series(k7) * 7) as i64;
        sum = sum - (arith_series(k21) * 21) as i64;
        out.push_str(&sum.to_string());
        out.push('\n');
    }
    print!("{out}");
}

fn arith_series(k: i32) -> i32 {
    return k * (k + 1) / 2
}